galley/queries.py

```diff
@@ -212,3 +212,55 @@ def get_raw_menu_data(dates: List[str],
                     else:
                         continue
     return response
+
+
+def get_ops_recipe_items_query(recipe_ids: List[str]) -> Operation:
+    query = Operation(Query)
+    query.viewer.recipes(where=FilterInput(id=recipe_ids)).__fields__('id', 'parentRecipeItems')
+    query.viewer.recipes.parentRecipeItems.__fields__('recipe')
+    query.viewer.recipes.parentRecipeItems.recipe.__fields__('recipeItems')
+    query.viewer.recipes.parentRecipeItems.recipe.recipeItems.__fields__('id', 'subRecipe', 'preparations')
+    query.viewer.recipes.parentRecipeItems.recipe.recipeItems.subRecipe.__fields__('id')
+    query.viewer.recipes.parentRecipeItems.recipe.recipeItems.preparations.__fields__('id', 'name')
+    return query
+
+
+def get_raw_recipe_items_data(recipe_ids: List) -> Iterable[List[Dict]]:
+    """
+    Returns a list of dictionaries containing the recipe items data for the
+    recipe ids. If there is no recipe items data for the recipe ids, returns None.
+
+    :param recipe_ids: The recipe ids for which the recipe items data is to be
+    fetched.
+    """
+    query = get_ops_recipe_items_query(recipe_ids=recipe_ids)
+
+    validated_response_data = validate_response_data(
+            make_request_to_galley(
+                op=query.__to_graphql__(auto_select_depth=3),
+                variables={'id': recipe_ids}),
+            'recipes')
+    return validated_response_data
+
+
+def get_untagged_core_recipe_item_ids(ids):
+    recipe_item_ids = []
+    throw_error = True
+    if ids is not None and len(ids) > 0:
```

Comment:
```suggestion
    if ids:
```
This simplified line will check for both cases.